### ISSUE TITLE
OpticalSystemModule.apply_bounds moves bounded variables outside their bounds

### ISSUE THREAD

**Issue by manuelFragata** [optiland/optiland#852]
**Describe the bug**

`OpticalSystemModule.apply_bounds()` clamps each parameter to the wrong range. The parameters live in scaled space and `var.bounds` is already scaled, but `apply_bounds` passes the bounds through `inverse_scale` before clamping. The scaled parameter ends up clamped to the physical bound values.

A thickness of 3.25896 mm with bounds [1, 10] becomes 20 mm after one call, without any optimizer step:

```python
import optiland.backend as be
from optiland.ml import OpticalSystemModule
from optiland.optimization import OptimizationProblem
from optiland.samples.objectives import CookeTriplet

be.set_backend("torch")
be.grad_mode.enable()

lens = CookeTriplet()
problem = OptimizationProblem()
problem.add_variable(lens, "thickness", surface_number=1, min_val=1.0, max_val=10.0)
problem.add_operand(operand_type="f2", target=50, weight=1, input_data={"optic": lens})

module = OpticalSystemModule(lens, problem)
print("thickness before:", lens.surfaces[1].thickness)
print("param:", module.params[0].item(), "var.bounds:", problem.variables[0].bounds)

module.apply_bounds()  # thickness is inside [1, 10], nothing to clamp
module()               # write the parameter back into the lens
print("param after apply_bounds:", module.params[0].item())
print("thickness after:", lens.surfaces[1].thickness)
```

Output on master (e962067b):

```
thickness before: 3.25896
param: -0.6741039752960205 var.bounds: (-0.9, 0.0)
param after apply_bounds: 1.0
thickness after: tensor(20., grad_fn=<DivBackward0>)
```

The parameter is -0.674, inside `var.bounds` (-0.9, 0.0). `apply_bounds` clamps it to [1.0, 10.0] instead, so it jumps to 1.0, which is a thickness of 20 mm.

In a training loop this happens after the first step. In the torch example from the JOSS paper (#846), the first `apply_bounds` call sets the six thickness parameters to 2.5, 4.0, 2.5, 4.0, 2.5 and 40.0 in scaled space, and the loss is NaN from step 1 on.

**Expected behavior**

`apply_bounds` clamps the parameter to `var.bounds` as returned, the way `TorchAdamOptimizer._apply_bounds` does.

**Cause**

`optiland/ml/wrappers.py`:

```python
min_val, max_val = var.bounds  # already scaled
min_val = (
    var.variable.inverse_scale(min_val) if min_val is not None else None
)
```

#485 fixed the same mismatch in `TorchAdamOptimizer` (#483) but did not change `OpticalSystemModule`. `test_sync_params_to_problem_and_bounds` in `tests/test_ml.py` asserts the wrong result, so CI passes.

**Environment**

- Optiland: master, e962067b
- torch 2.13.0 (CPU), Python 3.13, Windows 11
